infinoted/infinoted-options.c

```diff
@@ -1003,6 +1013,7 @@ infinoted_options_free(InfinotedOptions* options)
   g_free(options->certificate_file);
   g_free(options->certificate_chain_file);
   g_free(options->root_directory);
+  inf_ip_address_free(options->listen_address);
```

Comment:
I was told that it is usual for functions that free memory to handle NULL themselves, therefore it might be a good idea to just do the same for the libinfinity free functions if they need it.